Course/Course/Data/Model/Data_CourseDates.swift

```diff
@@ -58,6 +58,100 @@ public extension DataLayer {
             case missedGatedContent = "missed_gated_content"
             case verifiedUpgradeLink = "verified_upgrade_link"
         }
+        
+        var status: BannerInfoStatus? {
+            if upgradeToCompleteGraded {
+                return .upgradeToCompleteGradedBanner
+            } else if upgradeToReset {
+                return .upgradeToResetBanner
+            } else if resetDates {
+                return .resetDatesBanner
+            } else if showDatesTabBannerInfo {
+                return .datesTabInfoBanner
+            }
+            
+            return nil
+        }
+        
+        // Cases are defied according to this link
+        // https://2u-internal.atlassian.net/browse/LEARNER-7724?focusedCommentId=479226
```

Comment:
This private link does not make sense for the Open edX repo and community.
Better to describe the entire flow here or move it to the documentation.

Reply:
Cases are self explanatory and I believe we can remove this reference link.